odak/learn/raytracing/boundary.py

```diff
@@ -220,7 +255,7 @@ def intersect_w_surface(ray, points):
     normal = get_triangle_normal(points)
     if len(ray.shape) == 2:
         ray = ray.unsqueeze(0)
-    if len(points) == 2:
+    if len(points.shape) == 2:
```

Comment:
It should be the tensor shape, not the length of tensor. I also fix same bug in "raytracing" module.